tests/test_main.py

```diff
@@ -2190,3 +2190,28 @@ class Model(BaseModel):
         "{'type': 'recursion_loop', 'loc': ('x', 0, 'x', 0), 'msg': "
         "'Recursion error - cyclic reference detected', 'input': {'x': [{...}]}}"
     )
+
+
+def test_protected_namespace_default():
+    with pytest.raises(NameError, match='Field "model_x" has conflict with protected namespace "model_"'):
+
+        class Model(BaseModel):
+            model_x: str
+
+
+def test_custom_protected_namespace():
```

Comment:
I would also tested that providing a custom namespace clears the default setting and `model_` would be allowed.